### PR TITLE
Fix: bug associated with setuptools_scm that prevented remote install & steamline version config

### LINKED ISSUE #331: Issue with installing stravalib remotely
version python3.10

 pip3 install git+https://github.com/stravalib/stravalib.git@pydantic_model

one issue raise error
```python 
import stravalib 

```
image

_Originally posted by @yihong0618 in https://github.com/stravalib/stravalib/issues/328#issuecomment-1417804578_

### PR DESCRIPTION
**Pull request by lwasser**
closes #331

## Description

This PR fixes a bug that @yihong0618 discovered when trying to install stravalib remotely from GitHub directly. Essentially it was throwing a circular import error which i think stemmed from the SCM version file not being built as expected in the remote install.

This change does a few things

1. it removes the `_version.py` file that i created to create a properly formatted version `v1.x` instead of `1.x`
2. Modifies scm settings to use a template to add the v. this allows us to remove the additional module mentioned above. 
3. Adds a try/except to `__init__.py` so that in the event that a failure does happen, it simply assigns the version attribute to be "unreleased"

## Type of change

Select the statement best describes this pull request.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update
- [ ] Other (please describe)

## Does your PR include tests

If you are fixing a bug or adding a feature, we appreciate (but do not require)
tests to support whatever fix of feature you're implementing.

- [ ] Yes
- [ ] No, i'd like some help with tests
- [ ] This change doesn't require tests

I'm not sure if we need to test this - we could write a small test that just created a `_version_generated.py` file and then deleted it for try and except.  

## Did you include your contribution to the change log?

- [x] Yes, `changelog.md` is up-to-date.

*If you are having a hard time getting the tests to run correctly feel free to
ping one of the maintainers here!*

Do we need tests for this? it's a remote install but we could test that the try/except works as expected. but really there's no reason for it not to work.

### CODE REVIEW

**Review by jsamoocha — APPROVED** [review]
Thanks for the quick resolution, LGTM.

File: pyproject.toml [hunk 1]
```diff
@@ -9,6 +9,8 @@ version_scheme =  "post-release"
 # (so there isn't a + in the version)
 local_scheme = "no-local-version"
 write_to =  "stravalib/_version_generated.py"
+write_to_template = '__version__ = "v{version}"'
```
Comment (yihong0618) [inline pyproject.toml:12]: cool

**Review by yihong0618 — APPROVED** [review]
LGTM thanks a lot

**Comment by lwasser** [comment]
ok y'all - thanks so much for the quick approval!! merging!

also @yihong0618 when you have a chance, are you open to sending me your environment to test?

`pip list --format=freeze > requirements.txt`
 I'd like to try to recreate your environment to see why it's still not building the scm file (just so i understand what might be going on for the future)!!

**Comment by jsamoocha** [comment]
FYI, I just rebased `pydantic_model` on `master` and force-pushed to include this change for facilitating future tests.

**Comment by yihong0618** [comment]
image

**Pull request merged**